Source/AwsCommonRuntimeKit/io/ClientBootstrap.swift

```diff
@@ -1,26 +1,26 @@
 //  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 //  SPDX-License-Identifier: Apache-2.0.
 import AwsCIo
-import Foundation
 
-public final class ClientBootstrap {
+public class ClientBootstrap {
```

Comment:
ooc why remove `final`?

Reply:
Any reason why we want to have final?

Reply:
The codebase had inconsistent final with some classes for no reason. We decided to remove that to keep it consistent.

Reply:
Ok gotcha, I think it can be boiled down to whether or not we want consumers to be able to subclass the class or not.
If yes -> don't use `final`
If no -> use `final

When in doubt, it's preferable to not support subclassing and use `final`. This has some nice, while minor, runtime performance boosts via static dispatch since the compiler knows the location of the methods since there are no subclasses.

Reply:
Please correct me if I am wrong. My understanding is that consumers can not subclass this unless we make this open explicitly.